diff --git a/drivers/media/platform/xilinx/Kconfig b/drivers/media/platform/xilinx/Kconfig
index a4bc592fdb3c..d13aa1e94b1d 100644
--- a/drivers/media/platform/xilinx/Kconfig
+++ b/drivers/media/platform/xilinx/Kconfig
@@ -8,6 +8,12 @@ config VIDEO_XILINX
 
 if VIDEO_XILINX
 
+config VIDEO_RCC_CAM
+       tristate "RC Car Control Camera driver"
+       depends on VIDEO_XILINX
+       ---help---
+           Driver for RC Car Control project (home project)
+
 config VIDEO_XILINX_CFA
 	tristate "Xilinx Video Color Filter Array"
 	depends on VIDEO_XILINX
diff --git a/drivers/media/platform/xilinx/Makefile b/drivers/media/platform/xilinx/Makefile
index a28dceabe906..6dc1c862180b 100644
--- a/drivers/media/platform/xilinx/Makefile
+++ b/drivers/media/platform/xilinx/Makefile
@@ -1,5 +1,6 @@
 xilinx-video-objs += xilinx-dma.o xilinx-vip.o xilinx-vipp.o
 
+obj-$(CONFIG_VIDEO_RCC_CAM) += xilinx-rcc-cam.o
 obj-$(CONFIG_VIDEO_XILINX) += xilinx-video.o
 obj-$(CONFIG_VIDEO_XILINX_CFA) += xilinx-cfa.o
 obj-$(CONFIG_VIDEO_XILINX_CRESAMPLE) += xilinx-cresample.o
